editor.planx.uk/src/pages/FlowEditor/lib/store/preview.ts

```diff
@@ -590,10 +433,121 @@ export const previewStore: StateCreator<
+            String(b.data?.val).split(".").length -
+            String(a.data?.val).split(".").length,
+        )
+        // Only keep options with a data value set (remove blanks)
+        .filter((option) => option.data?.val);
+
+      // Get existing passport values that match this node's fn
+      let passportValues = computePassport()?.data?.[data?.fn]?.sort();
+      if (!passportValues) return;
+      if (!Array.isArray(passportValues)) passportValues = [passportValues];
+
+      // Only proceed if at least one option's data val startsWith an existing passport value (eg passport retains most granular value only)
+      passportValues = passportValues.filter((passportValue: any) =>
+        sortedOptions.some((option) =>
+          passportValue.startsWith(option.data?.val),
+        ),
+      );
+      if (!passportValues.length) return;
+
+      // For each sorted option, check if it has a direct match in the passport
+      sortedOptions.forEach((option) => {
+        passportValues.forEach((passportValue: any) => {
+          if (option.data?.val === passportValue) {
+            if (option.id) optionsThatCanBeAutoAnswered.push(option.id);
+          }
+        });
+      });
+
+      // If we haven't found any exact matches, see if the passport has a more granular version of the option
+      if (optionsThatCanBeAutoAnswered.length === 0) {
+        sortedOptions.forEach((option) => {
+          passportValues.forEach((passportValue: any) => {
+            // TODO - respect dot-separated segments ??
+            if (passportValue.startsWith(option.data?.val)) {
+              if (option.id) optionsThatCanBeAutoAnswered.push(option.id);
+            }
+          });
+        });
+      }
+
+      // TODO - Handle blanks & "_nots" for Questions & Checklists
+      //   Different than Filters, these types _can_ be put to user and we need to decide when
+    }
+
+    // Filters auto-answer based on a heirarchy of collected flags
+    if (type === TYPES.Filter) {
+      // TODO - It's still only possible to filter on the default flagset category, in future category configured on node by editor?
+      const possibleFlags = flatFlags.filter(
+        (flag) => flag.category === DEFAULT_FLAG_CATEGORY,
+      );
+      const possibleFlagValues = possibleFlags.map((flag) => flag.value);
+      if (!possibleFlagValues) return;
+
+      // Get all flags collected so far based on selected answers, excluding flags not in the specified category
+      const collectedFlags: Flag[] = [];
+      Object.entries(breadcrumbs).forEach(([_nodeId, crumb]) => {
+        if (crumb.answers) {
+          crumb.answers.forEach((answerId) => {
+            const node = flow[answerId];
+            if (node.data?.flag && possibleFlagValues.includes(node.data.flag))
+              collectedFlags.push(node.data?.flag);
+          });
+        }
+      });
+      if (!collectedFlags) return;
```

Comment:
Please note that I'm specifically _not_ relying on the existing `collectedFlags` store method here - as that's one to revisit soon when trying to refactor filters more broadly ! If/when `collectedFlags` is also properly refactored, we can replace this - but this implmentation feels simpler & more readable for now